packages/nc-gui/composables/useProject.ts

```diff
@@ -1,31 +1,53 @@
 import type { MaybeRef } from '@vueuse/core'
-import { SqlUiFactory } from 'nocodb-sdk'
 import type { OracleUi, ProjectType, TableType } from 'nocodb-sdk'
-import { useNuxtApp, useRoute } from '#app'
-import type { ProjectMetaInfo } from '~/lib'
+import { SqlUiFactory } from 'nocodb-sdk'
+import { isString } from '@vueuse/core'
+import {
+  USER_PROJECT_ROLES,
+  computed,
+  createEventHook,
+  ref,
+  useApi,
+  useGlobal,
+  useInjectionState,
+  useNuxtApp,
+  useRoute,
+  useState,
+  useTheme,
+  watch,
+} from '#imports'
+import type { ProjectMetaInfo, Roles } from '~/lib'
 import type { ThemeConfig } from '@/composables/useTheme'
-import { createEventHook, useInjectionState } from '#imports'
 
 const [setup, use] = useInjectionState((_projectId?: MaybeRef<string>) => {
-  const { $api, $e } = useNuxtApp()
+  const { $e } = useNuxtApp()
+
+  const { api, isLoading } = useApi()
```

Comment:
You are using the same `api` reference for loading the table list as well, so if you try to reload the table list(context menu in the tree view) it may lead to some issues I think.

Reply:
What's the difference between using `$api` and `api` in this case though?
It would still be only one api instance that's used to fetch the data, regardless if it's the global one or a new one locally.

If there is/was an issue with the reload, it was a pre-existing condition 😄

Reply:
> What's the difference between using `$api` and `api` in this case though? It would still be only one api instance that's used to fetch the data, regardless if it's the global one or a new one locally.
> 
> If there is/was an issue with the reload, it was a pre-existing condition 😄

We are using the `isLoading` variable for toggling between loader and spinner.

```vue
 <div class="w-full min-h-[300px] flex-auto">
        <NuxtPage v-if="!isLoadingProject" />
        <div v-else class="w-full h-full flex justify-center items-center">
          <a-spin size="large" />

```

Reply:
global isLoading is *always* triggered if an api instance does a request.

So if `const { isLoading } = useApi()` is `true` then `const { isLoading } = useGlobal()` is true as well, at least that is the intended way ^^

Reply:
I can also confirm I used useNuxtApp -> $api to not trigger isLoading state while using API. (on project list page, color picker api, as pageLoading for every call was not ideal)

Reply:
I think earlier I'd used `useApi()` only for loading project and for rest of them the global `$api` is used

Reply:
As I suspected, the issue(s) come from something else. Will push changes in a bit ^^

Reply:
Fix pushed, I had to play around a bit to get it right
* Subpages should now wait (and show a spinner) until the parent pages are done fetching the project/tables etc.
* The reload issue in the video came from using `v-if` on a NuxtPage component, which doesn't work, when the condition turns from true to false, it crashes the app